### DISCUSSION TITLE
Why do most components have "use client"?

### DISCUSSION THREAD

**Discussion by kondo0602** [shadcn-ui/ui#1413] [Q&A] [answered]
Most components have "use client", is it necessary?

Some components seem to have "use client" even though they do not have event listeners or state.

For example, [the label component](https://github.com/shadcn-ui/ui/blob/613ec3583ff20c47b5a1a9d0ba91fafbc96b9226/apps/www/registry/default/ui/label.tsx).

**Comment by zeekrey** [comment, marked as answer]
The `useClient` is essential whenever an event listener is utilized - just as you mentioned. If you take a glance at [the radix label primitive](https://github.com/radix-ui/primitives/blob/main/packages/react/label/src/Label.tsx), you'll notice that it employs `onMouseDown`.

It's likely that the majority of primitives employ various event listeners, which is why the wrapper component includes the `useClient` directive.

**Reply by kondo0602** [reply]
I understood that the hooks are not exposed on this code but are used in radix-ui.

Thanks!

**Comment by JamesSingleton** [comment]
The `'use client'` at the top is because he is using Next.js and the app directory in his website and the template that you can use as a starting off point. You can read about it on Next.js' website https://nextjs.org/docs/app/building-your-application/rendering/client-components#using-client-components-in-nextjs

**Reply by kondo0602** [reply]
Thanks!
